src/layer/marker.ts

```diff
@@ -214,6 +214,7 @@ export class Marker extends Layer<DivIconMarker> {
             );
             return;
         }
+        const marker = markerIcon.markerIcon;
         const icon = markerDivIcon(this.map.plugin.parseIcon(marker));
```

Comment:
This seems unnecessary if the types are accurate; we already have a MarkerIcon and Icon. `parseIcon` returns a MarkerIcon from an Icon. But `parseIcon` also seems to be setting some defaults in case the properties don't already exist on the object, so I left it here.

`parseIcon` also includes a MarkerDivIcon on it's return value...